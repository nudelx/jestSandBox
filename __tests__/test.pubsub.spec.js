import EventSys from '../hadar/pubsub'
import ProductGallery from '../hadar/someComponent'
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
/// here you say that it should use the mock !!!!
jest.mock('../hadar/pubsub')

Enzyme.configure({ adapter: new Adapter() })

describe('mock test', () => {
  it('bla', () => {
    // console.log('EventSystem', EventSys)
    EventSys.subscribe({ type: 'this is fucking ugly code !!!' })
    expect(EventSys.subscribe.mock.calls).toMatchSnapshot()
    expect(1).toBe(1)
  })
})

describe('component using the EventSystem', () => {
  it('it should subscribe on mount ', () => {
    const component = shallow(<ProductGallery />) /// here it will be mounted so you can check how it was subscribed
    expect(toJson(component)).toMatchSnapshot()
    expect(EventSys.subscribe.mock.calls).toMatchSnapshot()
  })
})
