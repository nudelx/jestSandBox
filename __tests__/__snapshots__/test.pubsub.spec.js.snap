// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component using the EventSystem it should subscribe on mount  1`] = `<div />`;

exports[`component using the EventSystem it should subscribe on mount  2`] = `
Array [
  Array [
    Object {
      "type": "this is fucking ugly code !!!",
    },
  ],
  Array [
    ProductGallery {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "setState": [Function],
      "state": Object {
        "desktop": true,
        "images": undefined,
        "productImageIndex": 0,
        "videoId": undefined,
        "videoPlaying": false,
      },
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": ReactShallowRenderer {
          "_context": Object {},
          "_element": <ProductGallery />,
          "_forcedUpdate": false,
          "_instance": [Circular],
          "_newState": null,
          "_rendered": <div />,
          "_rendering": false,
          "_updater": [Circular],
        },
      },
      Symbol(enzyme.__setState__): [Function],
    },
    "Resize",
  ],
]
`;

exports[`mock test bla 1`] = `
Array [
  Array [
    Object {
      "type": "this is fucking ugly code !!!",
    },
  ],
]
`;
